File: 18- Gato/vb/GatoVB/GatoVB/Form1.vb
Public Class frmGatoVB

    Dim botonUno, botonDos, botonTres, botonCuatro, botonCinco, botonSeis, botonSiete, botonOcho, botonNueve, nadieGana As Boolean
    Dim turno, numeroRandom As Integer

    Private Sub btnSiete_Click(sender As Object, e As EventArgs) Handles btnSiete.Click
        If turno = 1 Then
            botonSiete = True
            btnSiete.Enabled = False
            lbl7.Text = "X"
            turno = 2
        Else
            botonSiete = True
            btnSiete.Enabled = False
            lbl7.Text = "O"
            turno = 1
        End If

    End Sub

    Private Sub btnOcho_Click(sender As Object, e As EventArgs) Handles btnOcho.Click
        If turno = 1 Then
            botonOcho = True
            btnOcho.Enabled = False
            lbl8.Text = "X"
            turno = 2
        Else
            botonOcho = True
            btnOcho.Enabled = False
            lbl8.Text = "O"
            turno = 1
        End If

    End Sub

    Private Sub btnNueve_Click(sender As Object, e As EventArgs) Handles btnNueve.Click
        If turno = 1 Then
            botonNueve = True
            btnNueve.Enabled = False
            lbl9.Text = "X"
            turno = 2
        Else
            botonNueve = True
            btnNueve.Enabled = False
            lbl9.Text = "O"
            turno = 1
        End If

    End Sub

    Private Sub btnGo_Click(sender As Object, e As EventArgs) Handles btnGo.Click
        If Not txtPlayer1.Text = "" And Not txtPlayer2.Text = "" Then

            Randomize()
            numeroRandom = Convert.ToInt32(Rnd() * 100)
            If numeroRandom <= 50 Then
                turno = 1
                MsgBox("Turno de " & txtPlayer1.Text)
            Else
                turno = 2
                MsgBox("Turno de " & txtPlayer2.Text)
            End If

            txtPlayer1.Enabled = False
            txtPlayer2.Enabled = False
            btnSeis.Enabled = True
            btnSiete.Enabled = True
            btnOcho.Enabled = True
            btnNueve.Enabled = True
            btnUno.Enabled = True
            btnDos.Enabled = True
            btnTres.Enabled = True
            btnCuatro.Enabled = True
            btnCinco.Enabled = True
            btnGo.Enabled = False

            tmrTiempo.Enabled = True
        End If
    End Sub

    Private Sub btnSeis_Click(sender As Object, e As EventArgs) Handles btnSeis.Click
        If turno = 1 Then
            botonSeis = True
            btnSeis.Enabled = False
            lbl6.Text = "X"
            turno = 2

        Else
            botonSeis = True
            btnSeis.Enabled = False
            lbl6.Text = "O"
            turno = 1

        End If


    End Sub

    Private Sub btnCinco_Click(sender As Object, e As EventArgs) Handles btnCinco.Click
        If turno = 1 Then
            botonCinco = True
            btnCinco.Enabled = False
            lbl5.Text = "X"
            turno = 2
        Else
            botonCinco = True
            btnCinco.Enabled = False
            lbl5.Text = "O"
            turno = 1
        End If


    End Sub

    Private Sub btnCuatro_Click(sender As Object, e As EventArgs) Handles btnCuatro.Click
        If turno = 1 Then
            botonCuatro = True
            btnCuatro.Enabled = False
            lbl4.Text = "X"
            turno = 2

        Else
            botonCuatro = True
            btnCuatro.Enabled = False
            lbl4.Text = "O"
            turno = 1
        End If

    End Sub

    Private Sub btnTres_Click(sender As Object, e As EventArgs) Handles btnTres.Click
        If turno = 1 Then
            botonTres = True
            btnTres.Enabled = False
            lbl3.Text = "X"
            turno = 2
        Else
            botonTres = True
            btnTres.Enabled = False
            lbl3.Text = "O"
            turno = 1

        End If

    End Sub

    Private Sub btnDos_Click(sender As Object, e As EventArgs) Handles btnDos.Click
        If turno = 1 Then
            botonDos = True
            btnDos.Enabled = False
            lbl2.Text = "X"
            turno = 2
        Else
            botonDos = True
            btnDos.Enabled = False
            lbl2.Text = "O"
            turno = 1

        End If

    End Sub

    Private Sub frmGatoVB_Load(sender As Object, e As EventArgs) Handles MyBase.Load

        btnSeis.Enabled = False
        btnSiete.Enabled = False
        btnOcho.Enabled = False
        btnNueve.Enabled = False
        btnUno.Enabled = False
        btnDos.Enabled = False
        btnTres.Enabled = False
        btnCuatro.Enabled = False
        btnCinco.Enabled = False

    End Sub


    'Contador
    Private Sub tmrTiempo_Tick(sender As Object, e As EventArgs) Handles tmrTiempo.Tick

        If lbl1.Text = "X" And lbl2.Text = "X" And lbl3.Text = "X" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer1.Text)
            Application.Restart()
        ElseIf lbl4.Text = "X" And lbl5.Text = "X" And lbl6.Text = "X" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer1.Text)
            tmrTiempo.Enabled = False
            Application.Restart()
        ElseIf lbl7.Text = "X" And lbl8.Text = "X" And lbl9.Text = "X" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer1.Text)
            tmrTiempo.Enabled = False
            Application.Restart()
        ElseIf lbl1.Text = "X" And lbl4.Text = "X" And lbl7.Text = "X" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer1.Text)
            tmrTiempo.Enabled = False
            Application.Restart()
        ElseIf lbl2.Text = "X" And lbl5.Text = "X" And lbl8.Text = "X" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer1.Text)
            tmrTiempo.Enabled = False
            Application.Restart()
        ElseIf lbl3.Text = "X" And lbl6.Text = "X" And lbl9.Text = "X" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer1.Text)
            tmrTiempo.Enabled = False
            Application.Restart()
        ElseIf lbl1.Text = "X" And lbl5.Text = "X" And lbl9.Text = "X" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer1.Text)
            tmrTiempo.Enabled = False
            Application.Restart()
        ElseIf lbl3.Text = "X" And lbl5.Text = "X" And lbl7.Text = "X" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer1.Text)
            tmrTiempo.Enabled = False
            Application.Restart()
        End If



        If lbl1.Text = "O" And lbl2.Text = "O" And lbl3.Text = "O" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer2.Text)
            Application.Restart()
        ElseIf lbl4.Text = "O" And lbl5.Text = "O" And lbl6.Text = "O" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer2.Text)
            tmrTiempo.Enabled = False
            Application.Restart()
        ElseIf lbl7.Text = "O" And lbl8.Text = "O" And lbl9.Text = "O" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer2.Text)
            tmrTiempo.Enabled = False
            Application.Restart()
        ElseIf lbl1.Text = "O" And lbl4.Text = "O" And lbl7.Text = "O" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer2.Text)
            tmrTiempo.Enabled = False
            Application.Restart()
        ElseIf lbl2.Text = "O" And lbl5.Text = "O" And lbl8.Text = "O" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer2.Text)
            tmrTiempo.Enabled = False
            Application.Restart()
        ElseIf lbl3.Text = "O" And lbl6.Text = "O" And lbl9.Text = "O" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer2.Text)
            tmrTiempo.Enabled = False
            Application.Restart()
        ElseIf lbl1.Text = "O" And lbl5.Text = "O" And lbl9.Text = "O" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer2.Text)
            tmrTiempo.Enabled = False
            Application.Restart()
        ElseIf lbl3.Text = "O" And lbl5.Text = "O" And lbl7.Text = "O" Then
            tmrTiempo.Enabled = False
            MsgBox("gano " & txtPlayer2.Text)
            tmrTiempo.Enabled = False
            Application.Restart()
        End If

        If Not lbl1.Text = "" And Not lbl2.Text = "" And Not lbl3.Text = "" And Not lbl4.Text = "" And Not lbl5.Text = "" And Not lbl6.Text = "" And Not lbl7.Text = "" And Not lbl8.Text = "" And Not lbl9.Text = "" Then

            If nadieGana = True Then
                tmrTiempo.Enabled = False
                MsgBox("Nadie gana")
                Application.Restart()
            End If
            nadieGana = True

        End If

        If turno = 1 Then
            lblPlayer1.BackColor = Color.Red
            lblPlayer2.BackColor = Color.White
        Else
            lblPlayer1.BackColor = Color.White
            lblPlayer2.BackColor = Color.Red
        End If


    End Sub



    Private Sub btnUno_Click(sender As Object, e As EventArgs) Handles btnUno.Click
        If turno = 1 Then
            botonUno = True
            btnUno.Enabled = False
            lbl1.Text = "X"
            turno = 2

        Else
            botonUno = True
            btnUno.Enabled = False
            lbl1.Text = "O"
            turno = 1

        End If

    End Sub
End Class

File: 18- Gato/cs/GatoCS/GatoCS/Form1.cs
using Microsoft.VisualBasic;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace GatoCS
{
	public partial class frmGatoCS : Form
	{



		private bool botonUno, botonDos, botonTres, botonCuatro, botonCinco, botonSeis, botonSiete, botonOcho, botonNueve, nadieGana;

		

		private int turno, numeroRandom;

		

		public frmGatoCS()
		{
			InitializeComponent();
		}


		//Inicio del form
		private void frmGatoCS_Load(object sender, EventArgs e)
		{
			btnSeis.Enabled = false;
			btnSiete.Enabled = false;
			btnOcho.Enabled = false;
			btnNueve.Enabled = false;
			btnUno.Enabled = false;
			btnDos.Enabled = false;
			btnTres.Enabled = false;
			btnCuatro.Enabled = false;
			btnCinco.Enabled = false;
		}


		//Timer
		private void tmrTiempo_Tick(object sender, EventArgs e)
		{
			if ((lbl1.Text ?? "") == "X" & (lbl2.Text ?? "") == "X" & (lbl3.Text ?? "") == "X")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer1.Text);
				Application.Restart();
			}
			else if ((lbl4.Text ?? "") == "X" & (lbl5.Text ?? "") == "X" & (lbl6.Text ?? "") == "X")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer1.Text);
				tmrTiempo.Enabled = false;
				Application.Restart();
			}
			else if ((lbl7.Text ?? "") == "X" & (lbl8.Text ?? "") == "X" & (lbl9.Text ?? "") == "X")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer1.Text);
				tmrTiempo.Enabled = false;
				Application.Restart();
			}
			else if ((lbl1.Text ?? "") == "X" & (lbl4.Text ?? "") == "X" & (lbl7.Text ?? "") == "X")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer1.Text);
				tmrTiempo.Enabled = false;
				Application.Restart();
			}
			else if ((lbl2.Text ?? "") == "X" & (lbl5.Text ?? "") == "X" & (lbl8.Text ?? "") == "X")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer1.Text);
				tmrTiempo.Enabled = false;
				Application.Restart();
			}
			else if ((lbl3.Text ?? "") == "X" & (lbl6.Text ?? "") == "X" & (lbl9.Text ?? "") == "X")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer1.Text);
				tmrTiempo.Enabled = false;
				Application.Restart();
			}
			else if ((lbl1.Text ?? "") == "X" & (lbl5.Text ?? "") == "X" & (lbl9.Text ?? "") == "X")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer1.Text);
				tmrTiempo.Enabled = false;
				Application.Restart();
			}
			else if ((lbl3.Text ?? "") == "X" & (lbl5.Text ?? "") == "X" & (lbl7.Text ?? "") == "X")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer1.Text);
				tmrTiempo.Enabled = false;
				Application.Restart();
			}

			if ((lbl1.Text ?? "") == "O" & (lbl2.Text ?? "") == "O" & (lbl3.Text ?? "") == "O")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer2.Text);
				Application.Restart();
			}
			else if ((lbl4.Text ?? "") == "O" & (lbl5.Text ?? "") == "O" & (lbl6.Text ?? "") == "O")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer2.Text);
				tmrTiempo.Enabled = false;
				Application.Restart();
			}
			else if ((lbl7.Text ?? "") == "O" & (lbl8.Text ?? "") == "O" & (lbl9.Text ?? "") == "O")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer2.Text);
				tmrTiempo.Enabled = false;
				Application.Restart();
			}
			else if ((lbl1.Text ?? "") == "O" & (lbl4.Text ?? "") == "O" & (lbl7.Text ?? "") == "O")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer2.Text);
				tmrTiempo.Enabled = false;
				Application.Restart();
			}
			else if ((lbl2.Text ?? "") == "O" & (lbl5.Text ?? "") == "O" & (lbl8.Text ?? "") == "O")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer2.Text);
				tmrTiempo.Enabled = false;
				Application.Restart();
			}
			else if ((lbl3.Text ?? "") == "O" & (lbl6.Text ?? "") == "O" & (lbl9.Text ?? "") == "O")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer2.Text);
				tmrTiempo.Enabled = false;
				Application.Restart();
			}
			else if ((lbl1.Text ?? "") == "O" & (lbl5.Text ?? "") == "O" & (lbl9.Text ?? "") == "O")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer2.Text);
				tmrTiempo.Enabled = false;
				Application.Restart();
			}
			else if ((lbl3.Text ?? "") == "O" & (lbl5.Text ?? "") == "O" & (lbl7.Text ?? "") == "O")
			{
				tmrTiempo.Enabled = false;
				MessageBox.Show("gano " + txtPlayer2.Text);
				tmrTiempo.Enabled = false;
				Application.Restart();
			}

			if (!string.IsNullOrEmpty(lbl1.Text) & !string.IsNullOrEmpty(lbl2.Text) & !string.IsNullOrEmpty(lbl3.Text) & !string.IsNullOrEmpty(lbl4.Text) & !string.IsNullOrEmpty(lbl5.Text) & !string.IsNullOrEmpty(lbl6.Text) & !string.IsNullOrEmpty(lbl7.Text) & !string.IsNullOrEmpty(lbl8.Text) & !string.IsNullOrEmpty(lbl9.Text))
			{
				if (nadieGana == true)
				{
					tmrTiempo.Enabled = false;
					MessageBox.Show("Nadie gana");
					Application.Restart();
				}

				nadieGana = true;
			}

			if (turno == 1)
			{
				lblPlayer1.BackColor = Color.Red;
				lblPlayer2.BackColor = Color.White;
			}
			else
			{
				lblPlayer1.BackColor = Color.White;
				lblPlayer2.BackColor = Color.Red;
			}
		}


		//Botones del gato
		private void btnUno_Click(object sender, EventArgs e)
		{
			if (turno == 1)
			{
				botonUno = true;
				btnUno.Enabled = false;
				lbl1.Text = "X";
				turno = 2;
			}
			else
			{
				botonUno = true;
				btnUno.Enabled = false;
				lbl1.Text = "O";
				turno = 1;
			}
		}

		private void btnDos_Click(object sender, EventArgs e)
		{
			if (turno == 1)
			{
				botonDos = true;
				btnDos.Enabled = false;
				lbl2.Text = "X";
				turno = 2;
			}
			else
			{
				botonDos = true;
				btnDos.Enabled = false;
				lbl2.Text = "O";
				turno = 1;
			}
		}

		private void btnTres_Click(object sender, EventArgs e)
		{
			if (turno == 1)
			{
				botonTres = true;
				btnTres.Enabled = false;
				lbl3.Text = "X";
				turno = 2;
			}
			else
			{
				botonTres = true;
				btnTres.Enabled = false;
				lbl3.Text = "O";
				turno = 1;
			}
		}

		private void btnCuatro_Click(object sender, EventArgs e)
		{
			if (turno == 1)
			{
				botonCuatro = true;
				btnCuatro.Enabled = false;
				lbl4.Text = "X";
				turno = 2;
			}
			else
			{
				botonCuatro = true;
				btnCuatro.Enabled = false;
				lbl4.Text = "O";
				turno = 1;
			}
		}

		private void btnCinco_Click(object sender, EventArgs e)
		{
			if (turno == 1)
			{
				botonCinco = true;
				btnCinco.Enabled = false;
				lbl5.Text = "X";
				turno = 2;
			}
			else
			{
				botonCinco = true;
				btnCinco.Enabled = false;
				lbl5.Text = "O";
				turno = 1;
			}
		}

		private void btnSeis_Click(object sender, EventArgs e)
		{
			if (turno == 1)
			{
				botonSeis = true;
				btnSeis.Enabled = false;
				lbl6.Text = "X";
				turno = 2;
			}
			else
			{
				botonSeis = true;
				btnSeis.Enabled = false;
				lbl6.Text = "O";
				turno = 1;
			}
		}

		private void btnSiete_Click(object sender, EventArgs e)
		{
			if (turno == 1)
			{
				botonSiete = true;
				btnSiete.Enabled = false;
				lbl7.Text = "X";
				turno = 2;
			}
			else
			{
				botonSiete = true;
				btnSiete.Enabled = false;
				lbl7.Text = "O";
				turno = 1;
			}
		}

		private void btnOcho_Click(object sender, EventArgs e)
		{
			if (turno == 1)
			{
				botonOcho = true;
				btnOcho.Enabled = false;
				lbl8.Text = "X";
				turno = 2;
			}
			else
			{
				botonOcho = true;
				btnOcho.Enabled = false;
				lbl8.Text = "O";
				turno = 1;
			}
		}

		private void btnNueve_Click(object sender, EventArgs e)
		{
			if (turno == 1)
			{
				botonNueve = true;
				btnNueve.Enabled = false;
				lbl9.Text = "X";
				turno = 2;
			}
			else
			{
				botonNueve = true;
				btnNueve.Enabled = false;
				lbl9.Text = "O";
				turno = 1;
			}
		}





		//Boton Go!
		private void btnGo_Click(object sender, EventArgs e)
		{
			if (!string.IsNullOrEmpty(txtPlayer1.Text) & !string.IsNullOrEmpty(txtPlayer2.Text))
			{
				VBMath.Randomize();
				numeroRandom = Convert.ToInt32(VBMath.Rnd() * 100);
				if (numeroRandom <= 50)
				{
					turno = 1;
					MessageBox.Show("Turno de " + txtPlayer1.Text);
				}
				else
				{
					turno = 2;
					MessageBox.Show("Turno de " + txtPlayer2.Text);
				}

				txtPlayer1.Enabled = false;
				txtPlayer2.Enabled = false;
				btnSeis.Enabled = true;
				btnSiete.Enabled = true;
				btnOcho.Enabled = true;
				btnNueve.Enabled = true;
				btnUno.Enabled = true;
				btnDos.Enabled = true;
				btnTres.Enabled = true;
				btnCuatro.Enabled = true;
				btnCinco.Enabled = true;
				btnGo.Enabled = false;
				tmrTiempo.Enabled = true;
			}
		}


	}
}
